lib/adminui.js

```diff
@@ -39,12 +40,24 @@ function resume(req, res, next) {
 }
 
 function setUfds(req, res, next) {
+    var user = req.session && req.session.data && req.session.data.user || 'none';
     var trySetUfds = function () {
         if (req.ufdsMaster.connected) {
             req.ufds = req.ufdsMaster;
+            delete ufdsWarnings[user];
             next();
         } else if (req.ufdsSlave.connected) {
+            if (!ufdsWarnings[user]) {
+                req.log.warn('Retrieving data from slave UFDS.');
+                ufdsWarnings[user] = true;
+            }
             req.ufds = req.ufdsSlave;
+            var isAuthRequest = req.route && req.route.name && req.route.name === 'authenticate';
```

Comment:
Please do not write code like this.  If you believe that `req.route.name` is supposed to be a string, please assert that it is and then just do the comparison.  For example:

```
assert.string(req.route.name, 'req.route.name');
var isAuthRequest = (req.route.name === 'authenticate');
```

Otherwise, you will silently fail if the rest of the program is broken or its structure changes.